pkg/eventprocessor/impl/executor/impl/txnscope.go

```diff
@@ -107,9 +108,16 @@ func (ts *txnScope) executeTxnEvents(
 				ErrorEventIdx: &idx,
 			}, nil
 		}
+
+		if res.TableID != nil {
```

Comment:
This is the core of the bug fix. Inside the loop of events that are being processed, we build a slice of `tableIds`. Before, we were simply getting the last table id, which is inside `res.TableID`